pkg/collector/horizontalpodautoscaler.go

```diff
@@ -28,16 +28,20 @@ import (
 	"github.com/open-telemetry/opentelemetry-operator/pkg/naming"
 )
 
-const defaultCPUTarget int32 = 90
-
 func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) client.Object {
 	autoscalingVersion := cfg.AutoscalingVersion()
 
 	labels := Labels(otelcol, cfg.LabelsFilter())
 	labels["app.kubernetes.io/name"] = naming.Collector(otelcol)
 
 	annotations := Annotations(otelcol)
-	cpuTarget := defaultCPUTarget
+	var cpuTarget int32
+	if otelcol.Spec.TargetCPUUtilization != nil {
+		cpuTarget = *otelcol.Spec.TargetCPUUtilization
+	} else {
+		cpuTarget = int32(90)
```

Comment:
I don't like that the default CPU target utilization is defined in two places: defaulting webhook and here. There should be only one place where defaults are defined - preferably in the defaulting webhook.